### PR TITLE
Remove area emulation from media-content:sync command

### LINKED ISSUE #1784: Remove area emulation from media-content:sync command
Remove area emulation from `\Magento\MediaContentSynchronization\Console\Command\Synchronize::execute`

### Testing
1. Add some images from media gallery to the content of category, product, page and block
2. Clear `media_content_asset` database table
3. Run `media-content:sync` command
4. Verify all Used In links are correctly displayed in the asset details

### PR DESCRIPTION
**Pull request by jmonteros422**
### Description (*)

This PR Removes area emulation from `\Magento\MediaContentSynchronization\Console\Command\Synchronize::execute`

### Fixed Issues (if relevant)

1. Fixes magento/adobe-stock-integration#1784: Remove area emulation from media-content:sync command

### Testing
1. Add some images from media gallery to the content of category, product, page and block
2. Clear `media_content_asset` database table
3. Run `media-content:sync` command
4. Verify all Used In links are correctly displayed in the asset details

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by jmonteros422** [comment]
@magento run all tests

**Review by BarnyShergold — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @BarnyShergold, thank you for the review.
[ENGCOM-8081](https://jira.corp.magento.com/browse/ENGCOM-8081) has been created to process this Pull Request
:eight_spoked_asterisk: @BarnyShergold, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Review by engcom-Bravo — COMMENTED** [review]
:x: QA Failed

The synchronization works properly only once. But if you clear the ``media_content_asset`` table twice, the Used in will not display any records
Manua testing steps: 
1. Add some images from media gallery to the content of category, product, page and block
2. Clear `media_content_asset` database table
3. Run `media-content:sync` command
4. Verify all Used In links are correctly displayed in the asset details
5. Clear `media_content_asset` database table one more time
6. Run `media-content:sync` command
7. Verify all Used In links are correctly displayed in the asset details

AR: The _Used in_ section is not displayed in image details

File: app/code/Magento/MediaContentSynchronization/Console/Command/Synchronize.php [hunk 1]
```diff
@@ -58,12 +48,7 @@ protected function configure()
     protected function execute(InputInterface $input, OutputInterface $output)
     {
         $output->writeln('Synchronizing content with assets...');
-        $this->state->emulateAreaCode(
```
Comment (sivaschenko) [inline app/code/Magento/MediaContentSynchronization/Console/Command/Synchronize.php:61]: This emulation is used for Magento to generate proper links when searching for content usages. When removing the emulation - the links generation should be fixed

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the PR @jmonteros422 ! Please see my comment

**Review by engcom-Bravo — CHANGES_REQUESTED** [review]
Request changes test

**Review by sivaschenko — APPROVED** [review]
Synchronization is not executed for entities that have not been updated since last sync execution, can you please update the entity before performing the second table cleanup and sync

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-8081](https://jira.corp.magento.com/browse/ENGCOM-8081) has been created to process this Pull Request
:eight_spoked_asterisk: @sivaschenko, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

Manual testing steps
1. Add some images from media gallery to the content of category, product, page and block
2. Clear `media_content_asset` database table
3. Run `media-content:sync` command
4. Verify all Used In links are correctly displayed in the asset details

The media content synchronization is covered in several scenarios on cucumber studio. There is no need to create a separate for this PR.

**Pull request merged**